[Windows] Kodi fails to build in Debug mode

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug
Here is a clear and concise description of what the problem is:

Kodi fails to build in Debug mode



## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Should be possible build in Debug mode



## Actual Behavior

`LINK : fatal error LNK1104: cannot open file 'V:\kodi\project\BuildDependencies\x64\lib\pcre2-8-static.lib' [V:\kodi-build\kodi.vcxproj]`





## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->
Rename `pcre2-8-static.lib` to `pcre2-8-staticd.lib` in some CMake place. File with correct name is generated but Visual Studio project contains a reference with no d suffix.


### To Reproduce
Steps to reproduce the behavior:

```
cmake -G "Visual Studio 17 2022" -A x64 -T host=x64 %path_source%

cmake --build . --config "Debug" > log.txt
```


### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
The debuglog can be found here:

https://paste.kodi.tv/idasifuguy.kodi



### Screenshots 
Here are some links or screenshots to help explain the problem:
<!--- Put your text below this line -->
![pcre2](https://github.com/xbmc/xbmc/assets/58434170/63251853-5ba8-47a8-b19e-c5f1c1b3653e)



## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- Put your text below this line -->



### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [X] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: latest master branch 8356d6baca545568dab60124f35edb6ed5560dc3



<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

--- a/cmake/modules/FindPCRE2.cmake
+++ b/cmake/modules/FindPCRE2.cmake
@@ -12,7 +12,7 @@ if(NOT TARGET ${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME})
   macro(buildPCRE2)
     set(PCRE2_VERSION ${${MODULE}_VER})
     if(WIN32)
-      set(PCRE_DEBUG_POSTFIX d)
+      set(PCRE2_DEBUG_POSTFIX d)
     endif()
 
     set(patches "${CORE_SOURCE_DIR}/tools/depends/target/${MODULE_LC}/001-all-enable_docs_pc.patch"
@@ -20,19 +20,23 @@ if(NOT TARGET ${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME})
 
     generate_patchcommand("${patches}")
 
+    if(CORE_SYSTEM_NAME STREQUAL darwin_embedded OR WINDOWS_STORE)
+      set(EXTRA_ARGS -DPCRE2_SUPPORT_JIT=OFF)
+    else()
+      set(EXTRA_ARGS -DPCRE2_SUPPORT_JIT=ON)
+    endif()
+
     set(CMAKE_ARGS -DBUILD_STATIC_LIBS=ON
+                   -DPCRE2_STATIC_PIC=ON
                    -DPCRE2_BUILD_PCRE2_8=ON
                    -DPCRE2_BUILD_PCRE2_16=OFF
                    -DPCRE2_BUILD_PCRE2_32=OFF
-                   -DPCRE2_SUPPORT_JIT=ON
+                   -DPCRE_NEWLINE=ANYCRLF
                    -DPCRE2_SUPPORT_UNICODE=ON
                    -DPCRE2_BUILD_PCRE2GREP=OFF
                    -DPCRE2_BUILD_TESTS=OFF
-                   -DENABLE_DOCS=OFF)
-
-    if(CORE_SYSTEM_NAME STREQUAL darwin_embedded)
-      list(APPEND CMAKE_ARGS -DPCRE2_SUPPORT_JIT=OFF)
-    endif()
+                   -DENABLE_DOCS=OFF
+                   ${EXTRA_ARGS})
 
     set(${CMAKE_FIND_PACKAGE_NAME}_COMPILEDEFINITIONS PCRE2_STATIC)
 

